cmd/mailbox-controller/controller.go

```diff
@@ -202,105 +191,116 @@ func (ctl *mbCtl) sync1(ctx context.Context, ref any) {
 	}
 }
 
+// sync returns true to retry, false on success or unrecoverable error
 func (ctl *mbCtl) sync(ctx context.Context, refany any) bool {
 	logger := klog.FromContext(ctx)
+	mbsName := ""
 	stName, ok := refany.(refSyncTarget)
 	if ok {
 		st, err := ctl.synctargetLister.Get(string(stName))
 		if err != nil {
 			logger.Error(err, "Failed to fetch SyncTarget from local cache", "stName", stName)
 			return false
 		}
-		mbwsName, err := ctl.mbwsNameOfSynctarget(st)
+		mbsName, err = ctl.mbsNameOfSynctarget(st)
 		if err != nil {
-			return true
+			if err.Error() == errNoSpaceId.Error() {
+				logger.V(4).Info("Failed to retrieve Space ID. Retrying", "syncTargetName", st.Name)
```

Comment:
The reader wonders what retrieval is being discussed here. I would make the error message a little more explanatory. Perhaps something like the following.

```suggestion
				logger.V(4).Info("Corresponding IS name not known yet. Retrying", "syncTargetName", st.Name)
```